### PR TITLE
Add sdk change for nrp version 2023-04-01

### PR DESCRIPTION
**Pull request by Joanna-Yang-Art**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by Joanna-Yang-Art** [comment]
@ArthurMa1978 . @archerzz and @ArcturusZhang Could you help us review and approve this PR?

**Review by archerzz — COMMENTED** [review]
@Joanna-Yang-Art Could you please fix the errors in the pipeline first? From the error logs, looks like there are 3 issues:
- spell check
- test error
- some implementation change is not align with API interface, need to update the interface. Run `eng\scripts\Export-API.ps1 {your_sdk}` to update it.

**Comment by ArthurMa1978** [comment]
@Joanna-Yang-Art the definition of property `httpListeners` here https://github.com/Azure/azure-rest-api-specs/blob/main/specification/network/resource-manager/Microsoft.Network/stable/2023-04-01/applicationGateway.json#L1657 is wrong, which is array of `ApplicationGatewayHttpListener` instead of a `SubResource`.

File: sdk/network/Azure.ResourceManager.Network/src/autorest.md [hunk 1]
```diff
@@ -471,6 +474,13 @@ directive:
   - from: networkInterface.json # a temporary fix for issue https://github.com/Azure/azure-sdk-for-net/issues/34094
     where: $.definitions.EffectiveNetworkSecurityGroup.properties.tagMap.type
     transform: return "object";
+  # Fix the httpListeners which is array of `ApplicationGatewayHttpListener` instead of a `SubResource`
```
Comment (ArthurMa1978) [inline sdk/network/Azure.ResourceManager.Network/src/autorest.md:477]: This is a workaround for the swagger spec issue

**Review by ArthurMa1978 — APPROVED** [review]

**Review by archerzz — APPROVED** [review]

**Comment by Joanna-Yang-Art** [comment]
Hi @ArthurMa1978 thanks for your help, we already realized the ApplicationGatewayHttpListener issue in swagger and pushed app gateway team to fix it. Our entire release plan got delayed by this. Could you help us hold this PR for now, once they checked in the fix to swagger repo, I will reach out to confirm with you see if the fix match with current sdk change or if there is anything need to be updated on sdk part

**Review by ArthurMa1978 — CHANGES_REQUESTED** [review]
Waiting for the swagger spec fixing

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**